### PR TITLE
Feature/vfox0.1 sprint2

### PR DESCRIPTION
**Pull request by pushpendranirwan**
Sprint 2 major feature for review :  
1. API for email verification.
2. Add 'favicon' for the site. It seems to be missing.
3. Code enhancement.
4. Start Webapp on default port 80.
5. All service provider list in a single template and provider description card modification.
6. Unit test cases addition.
7. Document deployment steps in installation.md.
8. All service provider list in a single template and provider description card modification.
9. Clicking on 'Learn More' should not require to 'allow popups'.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -40,60 +30,49 @@ To download the MySQL Community Edition image, run this command:
 
     docker pull mysql/mysql-server:tag
                 
-             
+Thetag is the label for the image version you want to pull (for example,5.5,5.6,5.7,8.0,
```
Comment (ajayvfox) [inline README.md:33]: Thetag -> The tag

File: README.md [hunk 2]
```diff
@@ -40,60 +30,49 @@ To download the MySQL Community Edition image, run this command:
...
+    a24888f0d6f4 mysql/mysql-server &quot;/entrypoint.sh my...&quot; 14 seconds ago Up
+    13 seconds (health: starting) 3306/tcp, 33060/tcp mysql1
 
-Substitute `newpassword` with the password of your choice. Once the password is reset, the server is ready for use.
+The container initialization might take some time. When the server is ready for use, theSTATUS of
+the container in the output of thedocker ps command changes from(health:
+starting) to(healthy).
 
-#### Container Shell Access
+The-d option used in thedocker run command above makes the container run in the background.
```
Comment (ajayvfox) [inline README.md:70]: The-d -> The -d

File: deployment.md [hunk 3]
```diff
@@ -0,0 +1,33 @@
+# VFOX 0.1 Deployment Guide
+
+### Purpose
+ Contain information about the installation process of backend as JAVA and frontend as Angular.
+### Prerequisites
+```
+Java 1.8+
+Mysql 5.x
+docker 1.3x
```
Comment (ajayvfox) [inline deployment.md:9]: Why do we need 'docker' on the instance?

File: deployment.md [hunk 4]
```diff
@@ -0,0 +1,33 @@
+# VFOX 0.1 Deployment Guide
+
+### Purpose
+ Contain information about the installation process of backend as JAVA and frontend as Angular.
+### Prerequisites
+```
+Java 1.8+
```
Comment (ajayvfox) [inline deployment.md:7]: We need to add commands to install all these.

File: vfox-restServer/src/main/java/com/vfoexchange/restServer/controller/UserController.java [hunk 5]
```diff
@@ -1,36 +1,182 @@
...
+        ResponseEntity<ResponseDTO> responseEntity;
+        if (userService.isAleadyExist(userDto.getUsername())) {
+            resp.setCode(HttpStatus.ALREADY_REPORTED.toString());
+            resp.setMsg("This email already exists in our system. Please try another email for registration");
+            responseEntity = new ResponseEntity<ResponseDTO>(resp, HttpStatus.OK);
+            LOGGER.info(userDto.getUsername()+" email already exists in our system.");
+            return responseEntity;
+        }
+        try {
+
```
Comment (ajayvfox) [inline vfox-restServer/src/main/java/com/vfoexchange/restServer/controller/UserController.java:47]: Remove unnecessary blank line.

File: vfox-restServer/src/main/java/com/vfoexchange/restServer/controller/UserController.java [hunk 6]
```diff
@@ -1,36 +1,182 @@
...
+            LOGGER.error("User " + e.getMessage());
+            resp.setCode(HttpStatus.BAD_REQUEST.toString());
+            resp.setMsg("Error occurred while sending mail.");
+            responseEntity = new ResponseEntity<ResponseDTO>(resp, HttpStatus.OK);
+        }
+
+        return responseEntity;
+    }
+
+
```
Comment (ajayvfox) [inline vfox-restServer/src/main/java/com/vfoexchange/restServer/controller/UserController.java:64]: Remove unnecessary blank line.

File: vfox-restServer/src/main/java/com/vfoexchange/restServer/controllerAdvice/CustomizedResponseEntityExceptionHandler.java [hunk 7]
```diff
@@ -0,0 +1,40 @@
+package com.vfoexchange.restServer.controllerAdvice;
+
```
Comment (ajayvfox) [inline vfox-restServer/src/main/java/com/vfoexchange/restServer/controllerAdvice/CustomizedResponseEntityExceptionHandler.java:2]: Remove unnecessary blank line.

File: vfox-restServer/src/main/java/com/vfoexchange/restServer/controllerAdvice/CustomizedResponseEntityExceptionHandler.java [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+    private static final Logger LOGGER = LoggerFactory.getLogger(CustomizedResponseEntityExceptionHandler.class);
+
+    @ExceptionHandler(UserNotFoundException.class)
+    public final ResponseEntity<ErrorDetails> handleUserNotFoundException(UserNotFoundException ex, WebRequest request) {
+        LOGGER.debug("handling 404 error on a user entry");
+        ErrorDetails errorDetails = new ErrorDetails(AppUtil.getDate(AppConstants.DATE_FORMAT), ex.getMessage(),
+                request.getDescription(false));
+        return new ResponseEntity<>(errorDetails, HttpStatus.NOT_FOUND);
+    }
+
```
Comment (ajayvfox) [inline vfox-restServer/src/main/java/com/vfoexchange/restServer/controllerAdvice/CustomizedResponseEntityExceptionHandler.java:31]: Remove unnecessary blank line.

File: vfox-restServer/src/main/java/com/vfoexchange/restServer/model/ServiceProviders.java [hunk 9]
```diff
@@ -0,0 +1,69 @@
+package com.vfoexchange.restServer.model;
+
```
Comment (ajayvfox) [inline vfox-restServer/src/main/java/com/vfoexchange/restServer/model/ServiceProviders.java:2]: Remove unnecessary blank line.

File: vfox-restServer/src/main/java/com/vfoexchange/restServer/model/Services.java [hunk 10]
```diff
@@ -0,0 +1,51 @@
+package com.vfoexchange.restServer.model;
+
```
Comment (ajayvfox) [inline vfox-restServer/src/main/java/com/vfoexchange/restServer/model/Services.java:2]: Remove unnecessary blank line.

File: vfox-restServer/src/main/java/com/vfoexchange/restServer/serviceImpl/EmailServicesImpl.java [hunk 11]
```diff
@@ -0,0 +1,62 @@
...
+import javax.mail.internet.MimeMessage;
+import java.io.UnsupportedEncodingException;
+import java.util.Properties;
+
+@Service("emailServices")
+public class EmailServicesImpl implements EmailServices {
+
+    @Autowired
+    private Environment environment;
+
```
Comment (ajayvfox) [inline vfox-restServer/src/main/java/com/vfoexchange/restServer/serviceImpl/EmailServicesImpl.java:25]: Remove unnecessary blank line.

File: vfox-restServer/src/test/java/com/vfoexchange/restServer/coreApp/VfoxRestSeverTests.java [hunk 12]
```diff
@@ -0,0 +1,86 @@
...
+
+       User user = new User();
+       user.setId(13);
+       user.setUsername("admin");
+       user.setFirstLogin(false);
+       user.setRoleId(1);
+       UserRole userRole = new UserRole();
+       userRole.setRole("advisor");
+
+          Mockito.when(userDao.findByUsernameWithState(Mockito.anyString())).thenReturn(user);
```
Comment (ajayvfox) [inline vfox-restServer/src/test/java/com/vfoexchange/restServer/coreApp/VfoxRestSeverTests.java:63]: Indentation is wrong.

File: vfox-restServer/src/main/java/com/vfoexchange/restServer/serviceImpl/EmailServicesImpl.java (hunk 11 above)
Comment (Yatin-Arora) [reply vfox-restServer/src/main/java/com/vfoexchange/restServer/serviceImpl/EmailServicesImpl.java:25]: Unnecessary blank line removed and committed.

File: vfox-restServer/src/main/java/com/vfoexchange/restServer/model/Services.java (hunk 10 above)
Comment (Yatin-Arora) [reply vfox-restServer/src/main/java/com/vfoexchange/restServer/model/Services.java:2]: Unnecessary blank line removed and committed.

File: vfox-restServer/src/main/java/com/vfoexchange/restServer/model/ServiceProviders.java (hunk 9 above)
Comment (Yatin-Arora) [reply vfox-restServer/src/main/java/com/vfoexchange/restServer/model/ServiceProviders.java:2]: Unnecessary blank line removed and committed.

File: vfox-restServer/src/main/java/com/vfoexchange/restServer/controllerAdvice/CustomizedResponseEntityExceptionHandler.java (hunk 8 above)
Comment (Yatin-Arora) [reply vfox-restServer/src/main/java/com/vfoexchange/restServer/controllerAdvice/CustomizedResponseEntityExceptionHandler.java:31]: Unnecessary blank line removed and committed.

File: vfox-restServer/src/main/java/com/vfoexchange/restServer/controllerAdvice/CustomizedResponseEntityExceptionHandler.java (hunk 7 above)
Comment (Yatin-Arora) [reply vfox-restServer/src/main/java/com/vfoexchange/restServer/controllerAdvice/CustomizedResponseEntityExceptionHandler.java:2]: Unnecessary blank line removed and committed.

File: vfox-restServer/src/main/java/com/vfoexchange/restServer/controller/UserController.java (hunk 6 above)
Comment (Yatin-Arora) [reply vfox-restServer/src/main/java/com/vfoexchange/restServer/controller/UserController.java:64]: Unnecessary blank line removed and committed.

File: vfox-restServer/src/main/java/com/vfoexchange/restServer/controller/UserController.java (hunk 5 above)
Comment (Yatin-Arora) [reply vfox-restServer/src/main/java/com/vfoexchange/restServer/controller/UserController.java:47]: Unnecessary blank line removed and committed.

File: vfox-restServer/src/test/java/com/vfoexchange/restServer/coreApp/VfoxRestSeverTests.java (hunk 12 above)
Comment (Yatin-Arora) [reply vfox-restServer/src/test/java/com/vfoexchange/restServer/coreApp/VfoxRestSeverTests.java:63]: Reformatted the code and committed.

File: README.md (hunk 1 above)
Comment (Yatin-Arora) [reply README.md:33]: Change done and committed.

File: README.md (hunk 2 above)
Comment (Yatin-Arora) [reply README.md:70]: Change done and committed.

File: deployment.md (hunk 3 above)
Comment (Yatin-Arora) [reply deployment.md:9]: Removed 'docker' as it is already been taken care in README.md file.
Earlier, it was included just as a Prerequisite.

File: deployment.md (hunk 4 above)
Comment (Yatin-Arora) [reply deployment.md:7]: Added install command for java 8 and nginx.
Removed Mysql and docker, as it is already been taken care in README.md file.

**Review by ajayvfox — APPROVED** [review]
LGTM

**Pull request merged**